.github/workflows/pre-release-pr.yml

```diff
@@ -0,0 +1,83 @@
+name: Release test pre-release version
+
+on:
+  repository_dispatch:
+    types: [pre-release-command]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.client_payload.pull_request.head.sha }}
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+      - name: Install deps
+        run: |
+          python -m pip install pip --upgrade
+          pip install poetry
+          pip install githubrelease
+          pip install autopub
+          pip install httpx
+      - name: Check if release exists
+        id: check_release
+        run: |
+          set +e
+          echo ::set-output name=release::$(autopub check)
+      - name: Metadata
+        id: metadata
+        if: steps.check_release.outputs.release == ''
+        run: echo "::set-output name=commit::$(git rev-parse HEAD)"
+      - name: Find Release Comment
+        uses: peter-evans/find-comment@v1
+        id: find_comment
+        if: steps.check_release.outputs.release == ''
+        with:
+          token: ${{ secrets.BOT_TOKEN }}
+          issue-number: ${{ github.event.client_payload.github.payload.issue.number }}
+          comment-author: botberry
+          body-includes: "# Pre-release"
+      - name: Create or update comment
+        uses: peter-evans/create-or-update-comment@v1
+        if: steps.check_release.outputs.release == ''
+        with:
+          token: ${{ secrets.BOT_TOKEN }}
+          comment-id: ${{ steps.find_comment.outputs.comment-id }}
+          issue-number: ${{ github.event.client_payload.github.payload.issue.number }}
+          body: |
+            # Pre-release
+            :wave:
+            Releasing commit [${{ steps.metadata.outputs.commit }}] to PyPi as pre-release! :package:
+          edit-mode: replace
+      - name: Publish pre-release version
+        if: steps.check_release.outputs.release == ''
+        id: release
+        env:
+          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
+        run: |
+          autopub prepare
+          poetry version $(poetry version -s)-dev.$(date '+%s')
+          poetry build
+          poetry publish --username __token__ --password ${{ secrets.PYPI_TOKEN }}
```

Comment:
Could we use the `POETRY_PYPI_TOKEN_PYPI` env var instead? I'm always hesitant to provide secrets in the command line. I know GitHub Actions attempts to obfuscate secretes, but I think it would still be better to use the env var

Reply:
Sure, I will also change it in the the other workflow if we are worried about leaking it